clang/lib/Driver/Driver.cpp

```diff
@@ -783,7 +783,14 @@ void Driver::CreateOffloadingDeviceToolChains(Compilation &C,
       if (SYCLTargetsValues->getNumValues()) {
         for (const char *Val : SYCLTargetsValues->getValues()) {
           llvm::Triple TT(Val);
-          if (TT.getArch() == llvm::Triple::UnknownArch || !TT.isSPIR()) {
+          // Check for invalid triple values.  For AOT, these evaluate to
+          // regular SPIR target, so catch these early.
+          StringRef A(TT.getArchName());
+          if (TT.getArch() == llvm::Triple::UnknownArch || !TT.isSPIR() ||
+              (TT.getSubArch() == llvm::Triple::NoSubArch &&
+               ((TT.getArch() == llvm::Triple::spir && !A.equals("spir")) ||
+                (TT.getArch() == llvm::Triple::spir64 &&
+                 !A.equals("spir64"))))) {
```

Comment:
In this case, relying on the `&&` prevalence and thus minimizing the bracing seems more readable to me

Reply:
True, it would make it more readable, but the build emits warnings if you don't wrap && within ||